--- a/tapset/linux/ipmib.stp
+++ b/tapset/linux/ipmib.stp
@@ -271,7 +271,7 @@
  * IPSTATS_MIB_INUNKNOWNPROTOS)
  */
 /* icmp_send() is called by ip_local_deliver_finish() */
-probe ipmib.InUnknownProtos=kernel.function("icmp_send")
+probe ipmib.InUnknownProtos=kernel.function("__icmp_send")
 {
 	skb = $skb_in;
 	op = 1;
@@ -360,7 +360,7 @@
  * counted in the global @ReasmTimeout (equivalent to SNMP's MIB
  * IPSTATS_MIB_REASMTIMEOUT)
  */
-probe ipmib.ReasmTimeout=kernel.function("icmp_send")
+probe ipmib.ReasmTimeout=kernel.function("__icmp_send")
 {
 	skb = $skb_in;
 	op = 0;
--- a/tapset/linux/linuxmib.stp
+++ b/tapset/linux/linuxmib.stp
@@ -30,7 +30,7 @@
 
 probe _linuxmib.DelayedACKs.A = kernel.function("tcp_send_ack")
 {
-	sk=$sk
+	sk=pointer_arg(1)
 	if ( !indelack_timer[sk] ) next
 	op=1
 	key = linuxmib_filter_key(sk,op);
--- a/testsuite/systemtap.examples/network/tcp_trace.stp
+++ b/testsuite/systemtap.examples/network/tcp_trace.stp
@@ -192,7 +192,7 @@
 	}
 }
 
-probe kernel.function("tcp_transmit_skb")
+probe kernel.function("__tcp_transmit_skb")
 {
 	sk = $sk
 	key = filter_key(sk)
